x-pack/plugins/upgrade_assistant/public/components/tabs/checkup/deprecations/reindex/flyout/checklist_step.tsx

```diff
@@ -0,0 +1,102 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import React, { Fragment } from 'react';
+
+import {
+  EuiButton,
+  EuiButtonEmpty,
+  EuiCallOut,
+  EuiFlexGroup,
+  EuiFlexItem,
+  EuiFlyoutBody,
+  EuiFlyoutFooter,
+  EuiSpacer,
+  EuiTitle,
+} from '@elastic/eui';
+import { ReindexStatus } from '../../../../../../../common/types';
+import { LoadingState } from '../../../../../types';
+import { ReindexState } from '../polling_service';
+import { ReindexProgress } from './progress';
+
+const buttonLabel = (status?: ReindexStatus) => {
+  switch (status) {
+    case ReindexStatus.failed:
+      return 'Try again';
+    case ReindexStatus.inProgress:
+      return 'Reindexing…';
+    case ReindexStatus.completed:
+      return 'Done!';
+    case ReindexStatus.paused:
+      return 'Resume';
+    default:
+      return 'Run reindex';
+  }
+};
+
+/**
+ * Displays a flyout that shows the current reindexing status for a given index.
+ */
+export const ChecklistFlyoutStep: React.StatelessComponent<{
+  closeFlyout: () => void;
+  reindexState: ReindexState;
+  startReindex: () => void;
+}> = ({ closeFlyout, reindexState, startReindex }) => {
+  const {
+    loadingState,
+    status,
+    reindexTaskPercComplete,
+    lastCompletedStep,
+    errorMessage,
+  } = reindexState;
+  const loading = loadingState === LoadingState.Loading || status === ReindexStatus.inProgress;
+
+  return (
+    <Fragment>
+      <EuiFlyoutBody>
+        <EuiCallOut
+          title="Index is unable to ingest, update, or delete documents while reindexing"
+          color="warning"
+          iconType="alert"
+        >
+          If you can’t stop document updates or need to reindex into a new cluster, consider using a
```

Comment:
We should probably include a blurb about keeping the Kibana instance up which initialized the re-index, otherwise, they will need to return to manually resume the steps.

Something to the effect of: 

> Staying on this page while processing a re-index is not necessary, however, if the Kibana instance goes down during the process you will need to return to resume it.